ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryPartition.java

```diff
@@ -64,12 +65,18 @@
 
   @Override
   public List<AmbryReplica> getReplicaIds() {
-    return clusterManagerCallback.getReplicaIdsForPartition(this);
+    return clusterManagerCallback.getReplicaIdsForPartition(this)
```

Comment:
Instead of copying the list and requiring a (potentially) unsafe cast. I think we could make more use of generics in `ClusterManagerCallback`.